core/src/main/scala/dagr/core/util/ReflectiveBuilder.scala

```diff
@@ -169,7 +164,7 @@ class ArgumentLookup[ArgType <: Argument](args: ArgType*) {
   def ordered : Seq[ArgType] = argumentDefinitions.toList.sortBy(_.index)
 
   /** Returns the ArgumentDefinition, if one exists, for the provided field name. */
-  def fieldFor(fieldName: String) : Option[ArgType] = this.byFieldName.get(fieldName)
+  def forField(fieldName: String) : Option[ArgType] = this.byFieldName.get(fieldName)
```

Comment:
I come back to the idea that it should read well in a sentence, e.g.:
```scala
val fieldName = "foo"
val arg = argumentLookup.fieldFor(fieldName)  // tells me this should return a Field object
val arg = argumentLookup.forField(fieldName) // tells me it should look something (an arg?) up for a field
val arg = argumentLookup.argFor(fieldName) // tells me it should look up at arg for this field name
```

The last two read sensibly to me, the first one doesn't.  I would characterize as:
1. Use `forP` when you want the _parameter type_ made clear - `P` is the parameter type
2. Use `rFor` when you want the _return type_ made clear - `r` is the thing being returned
3. Use `rForP` when you want to make both the return type and parameter type clearer